sdk/go/pulumi/types_builtins.go

```diff
@@ -505,20 +505,20 @@ func (*asset) ElementType() reflect.Type {
 	return assetType
 }
 
-func (in *asset) ToAssetOutput() AssetOutput {
-	return ToOutput(in).(AssetOutput)
+func (a *asset) ToAssetOutput() AssetOutput {
```

Comment:
Was this change run through generate.go? I suspect we'll need to revert here unless you've special cased in generate.go